resources/assets/lib/chat/conversation-view.tsx

```diff
@@ -116,23 +127,33 @@ export default class ConversationView extends React.Component<Props> {
       return;
     }
 
+    const uiState = this.currentChannel.uiState;
+
     if (this.didSwitchChannel) {
-      if (this.unreadMarkerRef.current) {
-        this.scrollToUnread();
-      } else {
-        this.scrollToBottom();
-      }
-      this.didSwitchChannel = false;
+      // This can happen after a turbolinks navigation,
+      // so we have to wait for the elements to be on the current document before scrolling.
+      core.reactTurbolinks.runAfterPageLoad(this.id, action(() => {
```

Comment:
also need listener removal for this

Looking again the `runAfterPageLoad` function should just return a disposer :thinking: then the id wouldn't be needed anymore. And looking even closer I noticed one is called synchronously and the other is not - that may cause confusing behavior difference.

Reply:
doesn't `.one()` unbind automatically after it's done? 🤔

Reply:
that's assuming the component is still mounted when it runs

Reply:
sounds like `runAfterPageLoad` should change; `$(document).off` in the component without `$(document).on` seems counter intuitive 🤔